Main.java

```diff
@@ -36,42 +38,80 @@ public static void main(String[] args) throws Exception {
                         System.out.println(argument + " is a shell builtin");
                     }
                     else {
-                        String filePath = fileExists(paths, argument);
+                        String filePath = findFileInPath(paths, argument);
                         if (filePath != null){
                             System.out.println(argument + " is " + filePath);
                         }
                         else {
-                            System.out.println(input + ": command not found");
+                            System.out.println(argument + ": not found");
                         }
                     }
+                    break;
+                case "pwd":
+                    System.out.println(System.getProperty("user.dir"));
+                    break;
+                case "cd":
+                    if (argument.charAt(0) == '~'){
+                        if (argument.length() == 1){
+                            argument = System.getenv("HOME");
+                        }
+                        else {
+                            argument = System.getenv("HOME") + argument.split("~")[0];
+                        }
+                    }
+                    File directory = new File(argument);
+                    String[] parentLevels = argument.split("/");
+                    boolean relativeDir = false;
+                    if (parentLevels[0].equals(".") || parentLevels[0].equals(".."))  {
+                        relativeDir = true;
+                        for (int i = 0; i < parentLevels.length; i++){
+                            String level = parentLevels[i];
+                            if (!level.equals("..")) {
+                                String dir = "";
+                                for (int j = i+1; j < parentLevels.length; j++){
+                                    dir = dir + "/" + parentLevels[j];
+                                }
+                                dir = System.getProperty("user.dir") + dir;
+                                System.setProperty("user.dir", dir);
+                                directory = new File(dir);
+                                break;
+                            }
+                            Path currentDirectory = Paths.get(System.getProperty("user.dir"));
+                            Path parentDirectory = currentDirectory.getParent();
```

Comment:
Path actually provides many of the things you are doing above. You should either do this entirely with strings (because you want to show you can), or do it entirely with Path (which is how you would do it for a job).